### PR TITLE
Functions for calculate lower and high range of an IP address using a CIDR mask

### PR DESCRIPTION
**Pull request by YiuRULE**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):

Functions for calculate the lower range and an higher range for an IP using a CIDR, both for `IPv4` and `IPv6`.

Detailed description (optional):

This two functions, named `IPv4CIDRtoIPv4Range` and `IPv6CIDRtoIPv6Range` take two arguments, an IP address (in `IPv4` address  for the `IPv4` function, and in `IPv6` address 
 for `IPv6` function), and an `UInt8` who correspond to the CIDR mask we want to apply.

An equivalent of  `IPv4CIDRtoIPv4Range(IPv4NumToString('192.168.0.0'), 16)` can be interpreted as `192.168.0.0/16` using the CIDR notation.

It return a tuple containing two `IPv4` for the `IPv4` functions and a tuple containing two `IPv6` for the `IPv6`, it contain for both the theorically the lowest and highest IP for the subnet.

This function can potentially use to know if an IP is part of a subnet, and also if we use the lowest or highest range to aggregate the IP by subnet.

A possible solution was posted on issue #247 for an `IPv4` range (solution based also on the first post of the issue) using directly a query and using a bit shifting. But it wouldn't be possible for calculate an `IPv6` subnet as Clickhouse stored `IPv6` using a `FixedString(16)`, who make the task a little bit difficult with doing a bit shifting on an array.

### CODE REVIEW

**Comment by geldot** [comment]
Would it make sense to return an (IPv4,IPv4) tuple rather than a (UInt32,UInt32) tuple? That would make the interface more intuitive for display/interactive use. Specific use cases can always cast to UInt32 afterwards if needed.

**Comment by YiuRULE** [comment]
Thanks for the suggestion, now both function return a tuple with an IPv4 and IPv6!

**Comment by alexey-milovidov** [comment]
Need to remove `.FunctionsCoding.h.swo` file.

**Comment by alexey-milovidov** [comment]
IPv6CIDRtoIPv6Range -> IPv6CIDRToRange

File: dbms/src/Functions/FunctionsCoding.h [hunk 1]
```diff
@@ -1450,4 +1452,252 @@ class FunctionToStringCutToZero : public IFunction
...
+        return static_cast<UInt32>(src_byte_shift | cidr_mask_byte_shift);
+    }
+
+    static constexpr auto name = "IPv4CIDRtoIPv4Range";
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionIPv4CIDRtoIPv4Range>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 2; }
+    bool isInjective(const Block &) override { return true; }
```
Comment (alexey-milovidov) [inline dbms/src/Functions/FunctionsCoding.h]: :-1:

**Comment by alexey-milovidov** [comment]
Missing performance test.

**Comment by alexey-milovidov** [comment]
I will fix all the remaining issues.

**Pull request merged**